lazy_theta_star_b/src/lazy_theta_star_b.cpp

```diff
@@ -0,0 +1,529 @@
+
+        costmap_->worldToMap(start.pose.position.x, start.pose.position.y, reinterpret_cast<unsigned int &>(src.x),
+                             reinterpret_cast<unsigned int &>(src.y));
+
+
+        costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, reinterpret_cast<unsigned int &>(dst.x),
+                             reinterpret_cast<unsigned int &>(dst.y));
+        src.x = int(src.x);
+        src.y = int(src.y);
+
+        dst.x = int(dst.x);
+        dst.y = int(dst.y);
+
+        data.clear();
+        pq.clear();
+        path.clear();
+        posn.clear();
+        lethal_cost = LETHAL_COST;
+
+        sizeX = costmap_->getSizeInCellsX();
+        sizeY = costmap_->getSizeInCellsY();
+
+        //has been added so that if the robot is stuck at a place with costmap cost higher than 127, then only for that run, the planner be able to give a path
+        if (costmap_->getCost(src.x, src.y) >= lethal_cost && costmap_->getCost(src.x, src.y) <= 253) {
+            lethal_cost = costmap_->getCost(src.x, src.y);
+        }
+        
+        initializePosn();
+        data.reserve(int((sizeX) * (sizeY) * 0.01));
+
+        clearRobotCell(src.x, src.y);
+
+         if (!isSafe(src.x, src.y) || !isSafe(dst.x, dst.y) || !withinLimits(src.x, src.y) ||
+            !withinLimits(dst.x, dst.y)) {
+            RCLCPP_INFO(node_->get_logger(), "THE SOURCE OR DESTINATION IS EITHER NOT WITHIN THE SIZE MAP OR IS OBSTRUCTED");
+            global_path.poses.clear();
+            return global_path;
+
+        }
+
+        data.push_back({(src.x), (src.y), 0, dist(src.x, src.y, dst.x, dst.y), 0, -1,
+                        dist(src.x, src.y, dst.x, dst.y)});
+
+        addIndex(src.x, src.y, 0);
+
+        pq.push_back({0, &(data[0].g), &(data[0].f)});
+        //added as a buffer, since my binary_heap_del_min starts from the index 1 instead of 0
+        pq.push_back({0, &(data[0].g), &(data[0].f)});
+
+
+        //The algorithm begins here
+        id curr_id = 0;
+        id id_gen = 1;
+        id last_id = 0;
+        int cy;
+        int cx;
+        cx = (data[curr_id].x);
+        cy = (data[curr_id].y);
+
+        if (!los_check(src.x, src.y, dst.x, dst.y)) {
+            
+            while (pq.size() > 1) {
+
+                data[curr_id].counter++;
+               
+                 int moves[8][2] = {{cx,     cy + 1},
+                                   {cx + 1, cy},
+                                   {cx,     cy - 1},
+                                   {cx - 1, cy},
+                                   {cx + 1, cy + 1},
+                                   {cx + 1, cy - 1},
+                                   {cx - 1, cy - 1},
+                                   {cx - 1, cy + 1}};
+
+                if (how_many_corners != 8) {
+                    how_many_corners = 4;
+                }
+
+                int mx, my;
+                id m_id;
+                id curr_par = (data[curr_id].parentId);
```

Comment:
Same here with references, make everything a reference you can do you don't copy information (as long as not copying is allowable, if you're going to modify it later and need to rely on the fact that its a deep copy, then do it this way, but I don't think you need to do that for Theta*)